addons/analytic/models/analytic_line.py

```diff
@@ -192,3 +193,18 @@ class AccountAnalyticLine(models.Model):
         [('other', 'Other')],
         default='other',
     )
+    fiscal_year = fields.Integer(
+        compute='_compute_fiscal_date',
+        search='_search_fiscal_date',
+    )
+
+    @api.depends('date')
+    def _compute_fiscal_date(self):
+        for line in self:
+            if line.date is not None:
+                line.fiscal_year = self.env.company.compute_fiscalyear_dates(line.date)['date_from'].year
+
+    def _search_fiscal_date(self, operator, value):
+        fiscalyear_date_range = self.env.company.compute_fiscalyear_dates(fields.Date.today())
+        year_diff = fiscalyear_date_range['date_from'].year - value
+        return [('date', operator, fiscalyear_date_range['date_from'] - relativedelta(years=year_diff))]
```

Comment:
You don't need `year_diff`
```suggestion
        return [('date', operator, fiscalyear_date_range['date_from'] + relativedelta(year=value))]
```